clinica/utils/exceptions.py

```diff
@@ -19,3 +19,23 @@ class ClinicaCAPSError(ClinicaException):
 
 class ClinicaParserError(ClinicaException):
     """Base class for parser errors."""
+
+
+class ClinicaInconsistentDatasetError(ClinicaException):
+    """Base class for inconsistent datasets errors."""
+
+    def __init__(self, cross_subj):
+        from clinica.utils.stream import cprint
+
+        max_subjects_displayed = 50
+        bound = min(len(cross_subj), max_subjects_displayed)
+        msg = (
+            f"{len(cross_subj)} subjects in your BIDS folder did not respect the longitudinal organisation "
+            f"from BIDS specification.\nThe subjects concerned are (showing only the first {bound}):\n\t- "
+        )
+        msg += "\n\t- ".join(cross_subj[:bound])
+        msg += (
+            f"\nClinica does not know how to handle cross sectional dataset, but it "
+            "can convert it to a Clinica compliant form (using session ses-M00)"
+        )
+        cprint(msg, lvl="warning")
```

Comment:
If you print the error as a warning here, it makes the error class behave in a very unexpected way. You simply need to format the message and call the parent constructor here.